clang/include/clang/Serialization/ASTReader.h

```diff
@@ -501,7 +502,7 @@ class ASTReader
   ///
   /// When the pointer at index I is non-NULL, the declaration with ID
   /// = I + 1 has already been loaded.
-  std::vector<Decl *> DeclsLoaded;
+  llvm::PagedVector<Decl *> DeclsLoaded;
```

Comment:
@zygoloid, @ChuanqiXu9 should we follow this pattern and make `IdentifiersLoaded`, `SelectorsLoaded`, `MacrosLoaded` and `SubmodulesLoaded` `llvm::PagedVector`s, too? They could benefit from the same improvements in larger workflows. Maybe outside of this PR.

Reply:
I actually tried to change those as well, however the gain is less clear, at least in my simple ROOT tests. I guess one would need better tuning of PageSize for those. Part of the problem is that using a pointer per page is a bit heavy. I have some ideas on how to reduce this, by introducing another layer of indirection, but definitely not for this PR.

Reply:
I think making this decision based on measurements of memory savings in practice makes a lot of sense.

Reply:
I will resolve this comment for now, and propose more conversions from std::vector to PagedVector with actual measurements of the gains.